pkg/cloudprovider/aws/subnets.go

```diff
@@ -0,0 +1,72 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/patrickmn/go-cache"
+	"go.uber.org/zap"
+)
+
+type SubnetProvider struct {
+	ec2api ec2iface.EC2API
+	cache  *cache.Cache
+}
+
+func NewSubnetProvider(ec2api ec2iface.EC2API) *SubnetProvider {
+	return &SubnetProvider{
+		ec2api: ec2api,
+		cache:  cache.New(CacheTTL, CacheCleanupInterval),
+	}
+}
+
+func (s *SubnetProvider) GetZonalSubnets(ctx context.Context, cluster *v1alpha1.ClusterSpec) (map[string][]*ec2.Subnet, error) {
```

Comment:
Looks like we use only the cluster name here, can we only pass in the name as opposed to passing in the complete spec?

Reply:
Yeah I went back and forth on this and I think I agree with you.